### PR TITLE
handle_comparison_with_dates

### PR DESCRIPTION
**Pull request by jgaleotti**
SQL do handle: 'MinorThanEquals' and other comparisons with dates.
It still does not support subqueries.

### CODE REVIEW

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/HeuristicsCalculator.java [hunk 1]
```diff
@@ -490,11 +499,35 @@ private Object getValue(Expression exp, DataRow data) {
...
+            str = removeQuotes(str);
+            return str;
         } else {
             cannotHandle(exp);
             return null;
         }
     }
+
+    /**
+     * Given a string "Hello World" returns "ello Worl".
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/HeuristicsCalculator.java:517]: this description is confusing... so it is not `removingQuotes` but always removing first and last char? if so, should change name of function (eg, `removeFirstAndLast`), or change semantics (ie, throw an execption if first and last are not quotes). also should it deal only with `"` or also `'`?

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/HeuristicsCalculator.java [hunk 2]
```diff
@@ -490,11 +499,35 @@ private Object getValue(Expression exp, DataRow data) {
...
+     * @return
+     */
+    private String removeQuotes(String str) {
+        if (str.length() < 2) {
+            throw new IllegalArgumentException("Cannot remove quotes from " + str);
+        }
+        return str.substring(1, str.length() - 1);
+    }
+
+    private boolean startsAndEndsWithQuotes(String str) {
```
Comment (arcuri82) [inline client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/HeuristicsCalculator.java:530]: just to clarify, does SQL treat `"` and `'` the same or differently?

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/HeuristicsCalculator.java (hunk 1 above)
Comment (jgaleotti) [reply client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/HeuristicsCalculator.java:517]: Changed to removeFirstAndLastCharacter()

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/HeuristicsCalculator.java (hunk 2 above)
Comment (jgaleotti) [reply client-java/controller/src/main/java/org/evomaster/client/java/controller/internal/db/HeuristicsCalculator.java:530]: yes. I added a test case to reflect that.

**Pull request merged**